### DISCUSSION TITLE
Calculating porosity should be an algorithm

### DISCUSSION THREAD

**Discussion by jgostick** [PMEAL/OpenPNM#1988] [Ideas]
Calculating porosity is annoyingly difficult to do since there are many nuances, like subtracting the intersection of pores and throats, dealing with the domain size, etc.  It's actually worth adding an algorithm under the new ``metrics`` module.  I will close #1070, since I'm 99% sure this is a better way.

**Comment by xu-kai-xu** [comment]
Can we just calculate the porosity using `throat.endpoints` as the boundaries between throats and pores? This may reduce the intersection of pores and throats. About the domain size, it may be hard to infer automatically.
